plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -796,4 +861,264 @@ public void testLimitPushdown()
         assertThat(query("SELECT string_col, long_col FROM " + ALL_TYPES_TABLE + "  WHERE int_col >0 AND bool_col = 'false' LIMIT " + MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
                 .isNotFullyPushedDown(LimitNode.class);
     }
+
+    @Test
+    public void testAggregationPushdown()
+    {
+        // Without the limit inside the passthrough query, pinot will only return 10 rows
+        assertThat(query("SELECT COUNT(*) FROM \"SELECT * FROM " + ALL_TYPES_TABLE + " LIMIT " + MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES + "\""))
+                .isFullyPushedDown();
+
+        // Test aggregates with no grouping columns
+        assertThat(query("SELECT COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE))
+                .isFullyPushedDown();
+
+        // Test aggregates with no grouping columns with a limit
+        assertThat(query("SELECT COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE +
+                "  LIMIT " + MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
+                .isFullyPushedDown();
+
+        // Test aggregates with no grouping columns with a filter
```

Comment:
Is the filter on `long_col` pushed down? We should have tests with both cases where the predicate gets pushed down and where the filtering is done by the engine.

Reply:
I thought this test compares the results with and without pushdown, verifies types and rows match. Let me know if you meant something else.

Reply:
I thought this test compares the results with and without pushdown, verifies types and rows match. Let me know if you meant something else.

Reply:
@elonazoulay The test disables ALL pushdowns. I meant a test where the predicate cannot be pushed down but the aggregation could be. The reasoning is that if the predicates are applied incorrectly or at the wrong time the aggregation would return different results.